### PR TITLE
Update all references to tux and Tux to tuxx and Tuxx.

### PR DESCRIPTION
**Pull request by Cheerazar**

### CODE REVIEW

File: Tuxedo.js [hunk 1]
```diff
@@ -62,6 +62,6 @@ Tux.Animations.Rotate.CounterClockRight = require('./Animations/Rotate/CounterCl
 Tux.Animations.Rotate.CounterClockUp = require('./Animations/Rotate/CounterClockUp');
 
 //warn user not to use this object in console
-console.warn('Using the Tux object directly is not recommended as it will mean loading all modules even those you don\'t intend to use.  Please review the TuxedoJS documents for requiring only the individual pieces of TuxedoJS that you will need.');
+console.warn('Using the Tuxx object directly is not recommended as it will mean loading all modules even those you don\'t intend to use.  Please review the TuxedoJS documents for requiring only the individual pieces of TuxedoJS that you will need.');
```
Comment (plauer) [inline Tuxedo.js]: We should discuss what we plan to do with the Tux vs. Tuxx naming conventions for variables.

File: docs/Overview.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
->`TuxedoJS` is a client-side CommonJS based web development framework built on Facebook's awesome `React` view-layer and `Flux` Architecture. It provides a semantic interface for working with `Flux` and augmented `React` components for managing different aspects of the view logic. `Tux` also provides some powerful prebuilt animation components that make animating transitions in `React`  a synch. Lastly, `TuxedoJS` leverages CommonJS modules to provide a completely modular framework which means, unlike other popular frameworks, you only load the pieces of `Tux` you actually intend to use.
+>`TuxedoJS` is a client-side CommonJS based web development framework built on Facebook's awesome `React` view-layer and `Flux` Architecture. It provides a semantic interface for working with `Flux` and augmented `React` components for managing different aspects of the view logic. `Tuxx` also provides some powerful prebuilt animation components that make animating transitions in `React`  a synch. Lastly, `TuxedoJS` leverages CommonJS modules to provide a completely modular framework which means, unlike other popular frameworks, you only load the pieces of `Tuxx` you actually intend to use.
```
Comment (plauer) [inline docs/Overview.md]: Got to this point b4 heading home.

File: README.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-![Tux Logo](https://raw.githubusercontent.com/TuxedoJS/TuxedoJS/master/Tux-full-logo.png "Tux-Logo") <br />
+![Tuxx Logo](https://raw.githubusercontent.com/TuxedoJS/TuxedoJS/master/Tux-full-logo.png "Tux-Logo") <br />
```
Comment (drabinowitz) [inline README.md]: Change "Tux-Logo" to "Tuxx-Logo"

File: Tuxedo.js [hunk 4]
```diff
@@ -1,39 +1,39 @@
 'use strict';
 
-//NOTE: please avoid using this object if possible as it requires in all tux modules, even ones you might not use.  Please review the documentation on this subject and only require in your desired tux modules
+//NOTE: please avoid using this object if possible as it requires in all tuxx modules, even ones you might not use.  Please review the documentation on this subject and only require in your desired tux modules
 var Tux = {};
```
Comment (drabinowitz) [inline Tuxedo.js]: Change object name to Tuxx.

File: __tests__/TuxModules-test.js [hunk 5]
```diff
@@ -2,85 +2,85 @@
```
Comment (drabinowitz) [inline __tests__/TuxModules-test.js]: Need to rename file.

File: docs/TuxActions.md [hunk 6]
```diff
@@ -1,4 +1,4 @@
-# Tux/Actions
```
Comment (drabinowitz) [inline docs/TuxActions.md]: Rename file.

File: docs/TuxModularity.md [hunk 7]
```diff
@@ -1,22 +0,0 @@
-# Tux Modularity
```
Comment (drabinowitz) [inline docs/TuxModularity.md]: Why is this file being removed.

File: docs/TuxxActionStores.md [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+The interface for creating `ActionStores` is exposed via a `createStore` constructor function on the `ActionStores` object.
+
+```javascript
+    var ActionStores = require('tuxx/Stores/ActionStores');
+```
+
+***
+
+### <a id="Creating-ActionStores"></a>2) Creating ActionStores [#](#Creating-ActionStores)
+`ActionStores` are an implementation on top of the basic `TuxStore`. Thus, they have all of the same properties. The only difference is that `ActionStores` expect to have a `register` method passed in. If this `register` key is defined, then the `ActionStore` will invoke the method with the store as the context and use the object the method returns to register the store with `TuxxActions`. Let's take a look at an example
```
Comment (drabinowitz) [inline docs/TuxxActionStores.md]: Change TuxStore to TuxxStore.

File: docs/TuxxActionStores.md [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+        CATEGORY1_ACTION2: this.ONANOTHERACTION,
+      },
+      CATEGORY2: {
+        ACTION1: this.ONFINALACTION
+      }
+    }
+```
+
+#### <a id="createStore-actionStore"></a>Return - `actionStore` - type: OBJECT [#](#createStore-actionStore)
+Returns a `TuxStore` instance.
```
Comment (drabinowitz) [inline docs/TuxxActionStores.md]: Change TuxStore to TuxxStore.

File: docs/TuxxActions.md [hunk 10]
```diff
@@ -0,0 +1,405 @@
...
+```
+
+Here is a complete example of the before method.
+
+```javascript
+    actionCategory.before('add', function (nextCallbackToInvokeBeforeDispatching, actionBody) {
+
+      console.assert(actionBody.text === 'text passed in from action or last before callback');
+
+      nextCallbackToInvokeBeforeDispatching({text: 'invoking dispatch or next callback with new body.
```
Comment (drabinowitz) [inline docs/TuxxActions.md]: Missing ' at end of string.

File: docs/TuxxActions.md [hunk 11]
```diff
@@ -0,0 +1,405 @@
...
+    messageActions.register(messageStore, {
+      get: messageStore.onGet,
+      create: messageStore.onCreate,
+      delete: messageStore.onDelete
+    });
+
+    //the messageActions category wraps all of the business logic associated with messages, fully encapsulating all of the work associated with an app action
+    messageActions.before('get', function (nextCallback, actionBody) {
+      ajaxRequest('/messages', 'GET').then(function (results) {
+        nextCallback(results); //dispatch results of Ajax
```
Comment (drabinowitz) [inline docs/TuxxActions.md]: Should be `results.data`.

File: src/TuxxAnimations.js [hunk 12]
```diff
@@ -1,10 +1,10 @@
 'use strict';
```
Comment (drabinowitz) [inline src/TuxxAnimations.js]: Need to update the `tuxxAnimationKey` on line 165.

**Pull request merged**